src/Cake.Common.Tests/Unit/Tools/GitLink/GitlinkRunnerTests.cs

```diff
@@ -0,0 +1,241 @@
+                fixture.Settings.SolutionFileName = "solution.sln";
+
+                // Then
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("\"c:/temp\" -f \"solution.sln\"", result.Args);
+            }
+
+            [Fact]
+            public void Should_Set_Configuration()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+                fixture.Settings.Configuration = "Release";
+
+                // Then
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("\"c:/temp\" -c \"Release\"", result.Args);
+            }
+
+            [Fact]
+            public void Should_Set_Platform()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+                fixture.Settings.Platform = "AnyCPU";
+
+                // Then
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("\"c:/temp\" -p \"AnyCPU\"", result.Args);
+            }
+
+            [Fact]
+            public void Should_Set_Branch()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+                fixture.Settings.Branch = "master";
+
+                // Then
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("\"c:/temp\" -b \"master\"", result.Args);
+            }
+
+            [Fact]
+            public void Should_Set_LogFilePath()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+                fixture.Settings.LogFilePath = @"/temp/log.txt";
+
+                // When
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("\"c:/temp\" -l \"/temp/log.txt\"", result.Args);
+            }
+
+            [Fact]
+            public void Should_Set_ShaHash()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+                fixture.Settings.ShaHash = "abcdef";
+
+                // Then
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("\"c:/temp\" -s \"abcdef\"", result.Args);
+            }
+
+            [Fact]
+            public void Should_Set_PdbDirectory()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+                fixture.Settings.PdbDirectoryPath = DirectoryPath.FromString("pdb/");
+
+                // Then
+                var result = fixture.Run();
+
+                // Then
+                Assert.Equal("\"c:/temp\" -d \"/Working/pdb\"", result.Args);
+            }
+
+            [Fact]
+            public void Should_Set_PowerShell_Switch()
+            {
+                // Given
+                var fixture = new GitLinkFixture();
+                fixture.Settings.UsePowerShellCommand = true;
```

Comment:
Maybe best to use UsePowerShell to stay equal to the terminology in GitLink, but I'll leave that up to you.